tests/cxx/using_unused.cc

```diff
@@ -0,0 +1,21 @@
+//===--- using_unused.cc - test input file for iwyu -----------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// Tests that when we import a name via a using statement, IWYU tells us we 
+// need to include the original thing we're referencing, despite the fact that
+// it's not actually used.
+
+#include <vector>
+using std::vector;
```

Comment:
Yes, I also considered that using things from std is a little heavy, but decided it was good exercise.  For example, the issue with allocator on libc++ might not have been caught otherwise.  That being said, I don't have a strong opinion one way or the other so I am happy to change it.